src/pybind/rados/rados.pyx

```diff
@@ -3417,6 +3426,74 @@ returned %d, but should return zero on success." % (self.name, ret))
             raise make_ex(ret, "Failed to notify %r" % (obj))
         return True
 
+    def aio_notify(self, obj: str,
+                   oncomplete: Callable[[Completion, int, Optional[List], Optional[List]], None],
+                   msg: str = '', timeout_ms: int = 5000) -> Completion:
+        """
+        Asynchronously send a rados notification to an object
+
+        :param obj: the name of the object to notify
+        :type obj: str
```

Comment:
if we choose to go with type annotations, there is no need to document the types of parameters or the type of return value any more.